www/js/app.js

```diff
@@ -998,16 +1001,36 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
             document.getElementById('articleContent').contentWindow.focus();
             $("#searchingArticles").show();
             if (dirEntry.isRedirect()) {
-                selectedArchive.resolveRedirect(dirEntry, readArticle);
+                // Update expectedArticleURLToBeDisplayed before redirect. 
+                selectedArchive.resolveRedirect(dirEntry, function (resolvedDirEntry) {
+                    expectedArticleURLToBeDisplayed = resolvedDirEntry.namespace + "/" + resolvedDirEntry.url;
```

Comment:
As said in my previous review, it would simply the code to move this line at the beginning of `readArticle` (and remove all the same ones that are right before calling `readArticle`).
Here, it would also avoid creating this new inline function. In fact, the original code could be left

Reply:
The reason why I do update here is that I have no places to do the param udpate before `resolveRedirect()` call the `readArticle()` with a new dirEntry in this file except using this new inline function.   Please let me know if there is any other places to do the update after calling the resolveRedirect() method.

Reply:
Thanks for the hints. 
I over conplicated my code. The udpates are duplicated if we put the early return in the asynchronous part.